psi4/src/psi4/fisapt/fisapt.cc

```diff
@@ -2230,6 +2230,442 @@ void FISAPT::disp(std::map<std::string, SharedMatrix> matrix_cache, std::map<std
+
+    // Finish omega terms
+    std::shared_ptr<Matrix> omega_ar(AJK_ar->clone());
+    omega_ar->zero();
+    omega_ar->add(AJK_ar);
+    omega_ar->add(BJK_ar);
+    omega_ar->scale(4);
+    
+    std::shared_ptr<Matrix> omega_as(AJK_as->clone());
+    omega_as->zero();
+    omega_as->add(AJK_as);
+    omega_as->add(BJK_as);
+    omega_as->scale(2);
+    
+    std::shared_ptr<Matrix> omega_br(AJK_br->clone());
+    omega_br->zero();
+    omega_br->add(AJK_br);
+    omega_br->add(BJK_br);
+    omega_br->scale(2);
+    
+    std::shared_ptr<Matrix> omega_bs(AJK_bs->clone());
+    omega_bs->zero();
+    omega_bs->add(AJK_bs);
+    omega_bs->add(BJK_bs);
+    omega_bs->scale(4);
+
+    D.reset();
+    Sbr.reset();
+    Sas.reset();
+    Cocc0AB.reset();
+    D_Ni_a.reset();
+    D_Ni_b.reset();
+    J_D_ia.reset();
+    K_D_ia.reset();
+    J_D_ib.reset();
+    K_D_ib.reset();
+    D_ia.reset();
+    D_ib.reset();
+    AJK.reset();
+    BJK.reset();
+    AJK_ar.reset();
+    AJK_as.reset();
+    AJK_br.reset();
+    AJK_bs.reset();
+    BJK_ar.reset();
+    BJK_as.reset();
+    BJK_br.reset();
+    BJK_bs.reset();
+
+    // => Memory <= //
+
+    std::vector<std::shared_ptr<Matrix> > Cs;
+    Cs.push_back(Cocc0A);
+    Cs.push_back(Cvir0A);
+    Cs.push_back(Cocc0B);
+    Cs.push_back(Cvir0B);
+    Cs.push_back(C1a);
+    Cs.push_back(C1b);
+    Cs.push_back(C2a);
+    Cs.push_back(C2b);
+    Cs.push_back(Ct_Kr);
+    Cs.push_back(Ct_Ks);
+
+    size_t max_MO = 0, ncol = 0;
+    for (auto& mat : Cs) {
+        max_MO = std::max(max_MO, (size_t)mat->ncol());
+        ncol += (size_t)mat->ncol();
+    }
+
+    // => Get integrals from DFHelper <= //
+    auto dfh(std::make_shared<DFHelper>(primary_, auxiliary));
+    dfh->set_memory(doubles_ - Cs[0]->nrow() * ncol);
+    dfh->set_method("DIRECT_iaQ");
+    dfh->set_nthreads(nT);
+    dfh->initialize();
+    dfh->print_header();
+
+    dfh->add_space("a", Cs[0]);
+    dfh->add_space("r", Cs[1]);
+    dfh->add_space("b", Cs[2]);
+    dfh->add_space("s", Cs[3]);
+    dfh->add_space("a1", Cs[4]);
+    dfh->add_space("b1", Cs[5]);
+    dfh->add_space("a2", Cs[6]);
+    dfh->add_space("b2", Cs[7]);
+    dfh->add_space("r1", Cs[8]);
+    dfh->add_space("s1", Cs[9]);
+
+    dfh->add_transformation("Aar", "a", "r");
+    dfh->add_transformation("Abs", "b", "s");
+    dfh->add_transformation("Bar", "a1", "r1");
+    dfh->add_transformation("Bbs", "b1", "s1");
+    dfh->add_transformation("Cbr", "b2", "r1");
+    dfh->add_transformation("Cas", "a2", "s1");
+    dfh->add_transformation("Das", "a1", "s1");
+    dfh->add_transformation("Dbr", "b1", "r1");
+    dfh->add_transformation("Ebs", "b2", "s1");
+    dfh->add_transformation("Ear", "a2", "r1");
+
+    dfh->transform();
+
+    C1a.reset();
+    C1b.reset();
+    C2a.reset();
+    C2b.reset();
+    Ct_Kr.reset();
+    Ct_Ks.reset();
+    Cs.clear();
+    dfh->clear_spaces();
+
+    // => Blocking <= //
+
+    long int overhead = 0L;
```

Comment:
Just checking through here, did you update these overhead terms? If I recall correctly, you need to handle additional 3-index tensors in memory than the non-sinf version.

Reply:
I had not changed them, but I don't that it needs to be changed actually. Because the F tensor comes from adding D and E, the conventional version needs to make the same number of tensors as the sinf version if I'm not mistaken.